pycbc/workflow/core.py

```diff
@@ -764,6 +764,139 @@ def _filename(self, ifo, description, extension, segment):
         return "%s-%s-%s-%s.%s" % (ifo, description.upper(), start, 
                                    duration, extension)  
     
+class NewFile(File):
+    '''
+    This class holds the details of an individual output file 
+    This file(s) may be pre-supplied, generated from within the workflow
+    command line script, or generated within the workflow. The important stuff
+    is:
+
+    * The ifo that the File is valid for
+    * The time span that the OutFile is valid for
+    * A short description of what the file is
+    * The extension that the file should have
+    * The url where the file should be located
+
+    An example of initiating this class:
+    
+    c = File("H1", "INSPIRAL_S6LOWMASS", segments.segment(815901601, 815902001), file_url="file://localhost/home/spxiwh/H1-INSPIRAL_S6LOWMASS-815901601-400.xml.gz" )
+
+    another where the file url is generated from the inputs:
+
+    c = File("H1", "INSPIRAL_S6LOWMASS", segments.segment(815901601, 815902001), directory="/home/spxiwh", extension="xml.gz" )
+    '''
+
+    def __init__(self, ifos, exe_name, segs, lfn,
+                 extension=None, directory=None, tags=None, 
+                 store_file=True, use_tmp_subdirs=False):
+        """
```

Comment:
Hi Alex,

Yes, at this point the "lfn" name is a misnomer.  I was trying to essentially cover the three cases you list, but at the point this is called we don't know whether the user provided an http pfn (from which we have to extract an lfn) or a path (from which we have to construct both).  I did test both cases and verified that the entry in the resulting dax is correct.  I was also trying to allow for the None case, but forgot to set the default value in the arg list (I think that's the case you asked about below).  I can take another pass to make it more clear that that's what it's doing, but if the constructor takes both a pfn and lfn (both optionally None) then the burden of figuring out which the user provided floats back up to the caller, so I'd like to make it one argument and have the class decide which it is.